### ISSUE TITLE
Memory leak in Castle.Core proxy generation

### ISSUE THREAD

**Issue by sonicmouse** [jacqueskang/IpcServiceFramework#16]
In the client code, method `private static IpcRequest GetRequest(Expression exp, MyInterceptor interceptor)` ([link](https://github.com/jacqueskang/IpcServiceFramework/blob/892559374978c6b9f420130a646d68b4e61543d9/src/JKang.IpcServiceFramework.Client/IpcServiceClient.cs#L68))

You are generating a new `Castle.Core` proxy generator in every request. This should only be done once and statically. So, just simply add this to the top of your file:

```
private static readonly ProxyGenerator _proxyGenerator;
static IpcServiceClient()
{
    _proxyGenerator = new ProxyGenerator();
}
```
then in the method above, just use the static `_proxyGenerator` instead of generating a new one each time.

```
TInterface proxy = _proxyGenerator.CreateInterfaceProxyWithoutTarget<TInterface>(interceptor);
Delegate @delegate = lamdaExp.Compile();
@delegate.DynamicInvoke(proxy);
```

I was making thousands of requests per minute and just watched the memory climb and climb... now, with the fix, it doesn't climb at all... AND is much faster!

thanks...

**Comment by jacqueskang** [comment]
Thanks, I'll integreate for next release

**Comment by jacqueskang** [comment]
Resolved in v1.0.4.3
